### PR TITLE
Fix time difference validation and display server errors

### LINKED ISSUE #23: The max time difference between start and end date is 24 hrs and not limit on how short it is
When there is les than 60 min between  start and end date in mapathon summary report, 24 hrs error message is shown, although the difference is less than 24 hrs

image

### PR DESCRIPTION
**Pull request by d-rita**
This PR fixes #23 and shows relevant error messages returned by the backend server when a request fails.

**Tasks done:**
- Add reusable validation functions for each form inputs
- Add tests for validation functions.

**How to test:**
- Create a `.env` file in the base directory.
- Install dependencies using `npm i`
- Add `REACT_APP_API_URL="https://osm-stats.hotosm.org"` and `HOST=127.0.0.1`.
- Run `npm start`. Go to `http://127.0.0.1:3000/#/mapathon-summary-report` to test the app. For example, start date and end date with a time difference of less than 59 minutes. 

**Screenshots:**
- Fetching results for a 6 minute duration mapathon summary report:
6 min

- display time in 12-hour clock format in date picker sections:
12 hour clock format

### CODE REVIEW

File: src/components/formResponse.js [hunk 1]
```diff
@@ -3,6 +3,50 @@ import { FormattedMessage } from 'react-intl';
 import { BanIcon } from '../assets/svgIcons';
 import messages from './messages';
 
+const ErrorMessage = ({message}) => {
+    if (message === "Missing fields") {
```
Comment (omranlm) [inline src/components/formResponse.js:7]: I believe we can support more error messages as they come,

**Review by omranlm — APPROVED** [review]

File: src/components/formResponse.js (hunk 1 above)
Comment (d-rita) [reply src/components/formResponse.js:7]: Do you mean error messages from the backend? Or filtering more errors that will be specified in the frontend as we go along @omranlm?

Comment (omranlm) [reply src/components/formResponse.js:7]: I mean the more errors support

**Review by ramyaragupathy — APPROVED** [review]
@d-rita - this works as expected.
* displays time in 24 hrs format
* throws error only if the time difference in greater than 24 hrs

**Pull request merged**